setup.cfg

```diff
@@ -0,0 +1,5 @@
+[pycodestyle]
+max-line-length=120
+
+[pydocstyle]
+add-ignore=D101,D102,D104,D203,D205,D208,D209,D213,D300,D400,D401,D403,D406,D407,D413,D415
```

Comment:
The other changes are fine. This one needs work. The error codes are very very specific because PLY has some idiosyncratic syntax that is required and not Python standard. If the codes in the configuration you propose can be applied in the exact same sets to the exact same files, this can be merged. If not, this can't be merged.